### DISCUSSION TITLE
Property query does not exist on ... in client

### DISCUSSION THREAD

**Discussion by isaacfink** [trpc/trpc#4850] [Q&A] [answered]
I am using trpc in an express/sveltekit application, I am getting this weird error in the client
```
Property 'query' does not exist on type 'DecoratedProcedureRecord<unknown>'
```

All the routes are inferred correctly but it doesn't have the query property

What I have tried so far

1. verified my backend is set up correctly
2. made sure strict:true in all tsconfig files
3. made sure the serve and client packages are matching version, at one point I also set everything to latest

Nothing worked so far, any ideas?

**Comment by isaacfink** [comment]
Here is how my backend is configured

t.ts
```
import { initTRPC } from "@trpc/server";
import { IContext } from "./context";

const t = initTRPC.context<IContext>().create();

const publicProcedure = t.procedure;
const adminProcedure = t.procedure;
const router = t.router;

export { t, publicProcedure, adminProcedure, router };
```

someRouter.ts
```
import { t } from "../../trpc/t";
import { idValidator } from "../../validators";
import { menuFilterValidator } from "./validators";
import { getMenuDetails } from "./views";

export const router = t.router({
  all: t.procedure.input(menuFilterValidator).query(({ input }) => {
    return {};
  }),
});
```

server.ts
```
import { z } from "zod";
import { assetsRouter, authRouter, menusRouter, usersRouter } from "../admin/routes";
import { router } from "./t";
export const appRouter = router({
  auth: authRouter,
  menus: menusRouter,
  assets: assetsRouter,
  users: usersRouter,
  // hello: t.procedure.input(z.object({ name: z.string() })).query(() => "hello world!!!"),
});
// export type definition of API
export type AppRouter = typeof appRouter;
```

I added the hello procedure for testing and it didn't work either

**Comment by isaacfink** [comment]
I added this minimal example just to test and it as well didn't work, I think it has to do with the client package not the server
```
export const t = initTRPC.create();

export const r = t.router({
  hello: t.procedure.query(() => {
    return "world";
  }),
});
// export type definition of API
export type AppRouter = typeof r;
```

**Comment by isaacfink** [comment]
I am importing my client like this
```
import type { AppRouter } from '../../../api/trpc/server';

export const client = createTRPCProxyClient<AppRouter>({
	links: [
		httpBatchLink({
			url: 'http://127.0.0.1:4000/trpc',
			headers: async () => {
				return {
					authorization: 'token'
				};
			}
		})
	]
});
```
There is no workspace or monorepo setup, just a relative path to another project

**Comment by Knollludger** [comment, marked as answer]
I faced a similar issue, however I was able to figure out the problem fortunately. The client package has a peer dependency, which is the server package. I got the same error when I had installed just the client package without the server package being installed.

**Reply by Knollludger** [reply]
@isaacfink

**Reply by isaacfink** [reply]
Thanks I will try that out

**Reply by ShacharHarshuv** [reply]
So you're supposed to install the server package for the client as well? Seems wrong. Is it supposed to be a dev dependency only? What if there are other dependencies in the server that influence the type inference? It seems illogical that the client package will need to install them too.
What is the best practice here?

**Reply by xyunsh** [reply]
I got the same error  even I have added client package & server package

**Comment by lobotomoe** [comment]
In my case, I just didn’t need to use `.query` (I should have removed it). 🤦 

(Important: server component)

**Reply by xyunsh** [reply]
same issue, any updated ?
